qCC/ccPrimitiveFactoryDlg.cpp

```diff
@@ -158,3 +175,46 @@ void ccPrimitiveFactoryDlg::setSpherePositionToOrigin()
 	spherePosYDoubleSpinBox->setValue(0);
 	spherePosZDoubleSpinBox->setValue(0);
 }
+
+
+void ccPrimitiveFactoryDlg::setCoordinateSystemBasedOnSelectedObject()
+{
+	ccHObject::Container selectedEnt = m_win->getSelectedEntities();
+	for (auto entity : selectedEnt)
+	{
+		csMatrixTextEdit->setPlainText(entity->getGLTransformationHistory().toString());
+	}
+}
+
+void ccPrimitiveFactoryDlg::onMatrixTextChange()
+{	
+	bool valid = false;
+	isCSMatrixValid(valid);
+	if (valid)
+	{
+		ccLog::Print("Valid ccGLMatrix");
+	}
+}
+
+void ccPrimitiveFactoryDlg::setCSMatrixToIdentity()
+{
+	csMatrixTextEdit->blockSignals(true);
+	csMatrixTextEdit->setPlainText("1.00000000 0.00000000 0.00000000 0.00000000\n0.00000000 1.00000000 0.00000000 0.00000000\n0.00000000 0.00000000 1.00000000 0.00000000\n0.00000000 0.00000000 0.00000000 1.00000000");
+	csMatrixTextEdit->blockSignals(false);
+}
+
+ccGLMatrix ccPrimitiveFactoryDlg::isCSMatrixValid(bool& valid)
```

Comment:
shouldn't it be called ```getCSMatrix(bool& valid)```?

Reply:
yes, much better, I had originally just intended to check the matrix but it required duplicating most of the function to get it later so I changed it up to return the matrix rather than the bool, but didn't update the name